### DISCUSSION TITLE
Have the $CI environment variable set by default

### DISCUSSION THREAD

**Discussion by h1nk** [community/community#26628] [Actions]
<p>Many continuous integration services like <a href="https://docs.gitlab.com/ce/ci/variables/predefined_variables.html#variables-reference" rel="nofollow noopener">GitLab</a> and <a href="https://docs.travis-ci.com/user/environment-variables/#default-environment-variables" rel="nofollow noopener">Travis-CI</a> have the $CI environment variable set in their execution environments. Some tools such as Python pipenv for example will <a href="https://github.com/pypa/pipenv/blob/207f2f565d2c4493bd8cdfb55a9b565aa58984fc/pipenv/environments.py#L27" rel="nofollow noopener">check</a> for the presence of the $CI variable and automatically <a href="https://pipenv.rtfd.io/en/latest/advanced/#pipenv.environments.PIPENV_NOSPIN" rel="nofollow noopener">configure settings</a> approriate for a CI runtime environment.</p>

**Comment by hugovk** [comment]
<p>+1</p>
<p>Some tests should only be run on CI, others never on CI. Some tests should only selected or not on certain CIs.</p>
<p>Please include one called CI plus another to identify as GitHub Actions.</p>
<p>Here’s a summary of the truthy env vars from some popular CIs:</p>
<ul>
<li>AppVeyor: CI, <span>APPVEYOR</span>
</li>
<li>Azure Pipelines: TF_BUILD</li>
<li>CircleCI: CI, CIRCLECI</li>
<li>GitLab CI/CD: CI, <span>CI_SERVER, GITLAB_CI</span>
</li>
<li>Travis CI: CI, CONTINUOUS_INTEGRATION, TRAVIS</li>
</ul>
<p>Docs:</p>
<ul>
<li>
<p><a href="https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables" rel="nofollow noopener">https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables</a></p>
</li>
<li>
<p><a href="https://docs.gitlab.com/ce/ci/variables/predefined_variables.html#variables-reference" rel="nofollow noopener">https://docs.gitlab.com/ce/ci/variables/predefined_variables.html#variables-reference</a></p>
</li>
<li>
<p><a href="https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops#predefined-variables" rel="nofollow noopener">https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops#predefined-variables</a></p>
</li>
<li>
<p><a href="https://docs.travis-ci.com/user/environment-variables/#default-environment-variables" rel="nofollow noopener">https://docs.travis-ci.com/user/environment-variables/#default-environment-variables</a></p>
</li>
<li>
<p><a href="https://www.appveyor.com/docs/environment-variables/" rel="nofollow noopener">https://www.appveyor.com/docs/environment-variables/</a></p>
</li>
</ul>

**Comment by wipe2238** [comment]
<p>We already have GITHUB_ACTIONS, GITHUB_WORKFLOW, GITHUB_WORKSPACE, GITHUB_SHA, and about a dozen of others to make sure we’re running under GitHub, so this part is covered IMHO. Only CI is missing :slight_smile:</p>

**Comment by hugovk** [comment]
<p>Thanks, CI would still be good.</p>
<p>Here’s the docs about those (and it’s GITHUB_ACTION without an S):</p>
<p><a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables</a></p>

**Comment by ethomson** [comment]
<p>There’s two distinct values here.  GITHUB_ACTION is the name of the action that’s running, if any (and there may not be one).  GITHUB_ACTIONS is always set to show that you’re running within GitHub Actions.</p>

**Comment by hugovk** [comment]
<p>Thanks for clarifying. Please could you add GITHUB_ACTIONS to the the docs?</p>
<p><a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables</a></p>

**Comment by ethomson** [comment]
<p>Yes, an update is coming soon.  Thanks!</p>

**Comment by usmonster** [comment, marked as answer]
<p>Seems this is done!</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables" target="_blank" rel="noopener nofollow ugc">Environment variables - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/actions/reference/environment-variables</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
